lib/tools/getYouTubeChannelInfo.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * YouTube Channel Information Tool
+ * 
+ * Single tool that handles both authentication checking and channel info fetching.
+ * Internally validates YouTube tokens and returns either:
+ * - Full channel information if authenticated
+ * - Authentication error/instructions if not authenticated
+ * 
+ * This replaces the need for separate checkYouTubeAccess calls.
+ */
+
+import { z } from "zod";
+import { tool } from "ai";
+import { YouTubeChannelInfoResult } from "@/types/youtube";
+import { validateYouTubeTokens } from "@/lib/youtube/token-validator";
+import { fetchYouTubeChannelInfo } from "@/lib/youtube/channel-fetcher";
+import { YouTubeErrorBuilder } from "@/lib/youtube/error-builder";
+
+// Zod schema for parameter validation
+const schema = z.object({
+  artist_id: z.string().describe("Artist ID to get YouTube channel information for. This tool handles authentication checking internally.")
+});
+
+const getYouTubeChannelInfoTool = tool({
+  description:
+    "Get YouTube channel information for a specific account. " +
+    "This tool automatically checks authentication status and either returns channel data or authentication instructions. " +
+    "No need to check authentication separately - this tool handles everything internally. " +
+    "Returns comprehensive channel data including statistics, thumbnails, and branding if authenticated, " +
+    "or clear authentication instructions if tokens are missing/expired. " +
+    "IMPORTANT: This tool requires the artist_id parameter. If you don't know the artist_id, ask the user or use the current artist's artist_id.",
```

Comment:
does this match the variable name in the system prompt?
- I recommend explicitly stating this variable should come from the system prompt. be sure the spelling is precisely the same.